pkg/envtest/crd.go

```diff
@@ -308,10 +308,7 @@ func renderCRDs(options *CRDInstallOptions) ([]runtime.Object, error) {
 		}
 
 		// If CRD already in the list, skip it.
-		if existsUnstructured(crds, crdList) {
-			continue
-		}
-		crds = append(crds, crdList...)
+		crds = append(crds, findNotExistingUnstructured(crds, crdList)...)
```

Comment:
I think we can keep a map of CRDs key'd by GVKN for deduping that will keep the code a lot more simpler. WDYT ?
It took me sometime to parse the current logic :)

Reply:
I initially uses 2 slices to keep some existing behavior: If there are more than one definition of the same CRD, the last one will be installed eventually.
But it seems we can still achieve it using a map by keeping the last seen definition.

After thinking about on this, I guess the right thing to do here probaby is to error out. 
Thoughts?

Reply:
Given that this is `envtest` I'm fine either way. We can potentially improve it later. Maybe adding a few comments explaining why we're doing this and how it might help readers.

Reply:
Changed it to a map.